moPepGen/aa/VariantPeptideIdentifier.py

```diff
@@ -72,41 +79,110 @@ def parse_variant_peptide_id(label:str) -> List[VariantPeptideIdentifier]:
+                orf_id = field
+
+            elif IdentifierType is FusionVariantPeptideIdentifier:
+                if field.startswith('1-') or field.startswith('2-'):
+                    which_gene, var_id = field.split('-', 1)
+                    which_gene = int(which_gene)
+                else:
+                    # SECT or W2F
+                    var_id = field
+                    which_gene = 0
+                if which_gene in var_ids:
+                    var_ids[which_gene].append(var_id)
+                else:
+                    var_ids[which_gene] = [var_id]
+
+            elif any(field.startswith(str(t)) for t in VariantPrefix.alt_translation()):
+                alt_ids.append(field)
+
+            elif any(field.startswith(str(t)) for t in VariantPrefix.ctbv()):
+                if 1 in var_ids:
+                    var_ids[1].append(field)
+                else:
+                    var_ids[1] = [field]
+
+        if IdentifierType is None:
+            if not var_ids and orf_id is not None:
+                x_id = fields[0]
+                if len(fields) > 1:
+                    gene_id = fields[1]
+                IdentifierType = _set_identifier_type(
+                    IdentifierType,
+                    NovelORFPeptideIdentifier
+                )
             else:
-                variant_id = BaseVariantPeptideIdentifier(x_id, var_ids, orf_id, index)
+                x_id = fields[0]
+                IdentifierType = _set_identifier_type(
+                    IdentifierType,
+                    BaseVariantPeptideIdentifier
+                )
+
+        if IdentifierType is NovelORFPeptideIdentifier:
+            variant_id = NovelORFPeptideIdentifier(
+                transcript_id=x_id,
+                gene_id=gene_id,
+                codon_reassigns=alt_ids,
+                orf_id=orf_id,
+                index=index,
+                is_protein_coding=(x_id in coding_txs)
+            )
+        elif IdentifierType is FusionVariantPeptideIdentifier:
+            variant_id = FusionVariantPeptideIdentifier(
+                fusion_id=x_id,
+                first_variants=var_ids.get(1, []),
+                second_variants=var_ids.get(2, []),
+                peptide_variants=var_ids.get(0, []),
+                orf_id=orf_id,
+                index=index
+            )
+        elif IdentifierType is CircRNAVariantPeptideIdentifier:
+            variant_id = CircRNAVariantPeptideIdentifier(
+                circ_rna_id=x_id,
+                variant_ids=sum(var_ids.values(), []) + alt_ids,
+                orf_id=orf_id,
+                index=index
+            )
+        elif IdentifierType is BaseVariantPeptideIdentifier:
+            variant_id = BaseVariantPeptideIdentifier(
+                transcript_id=x_id,
+                variant_ids=sum(var_ids.values(), []) + alt_ids,
+                orf_id=orf_id,
+                index=index,
+                gene_id=gene_id
+            )
```

Comment:
this giant chuck of code isn't very elegent XD but I see how the header complexity has increased

Reply:
Let me add some inline docs. This is not as bad as it looks, and actually much better than before

Reply:
As mentioned below, the FASTA headers now can have the 5 variables. So now I first assign each field into the 5 variables, and then infer the correct FASTA header type (fusion, or circRNA, or variant or novelORF) based on those variables.

https://github.com/uclahs-cds/package-moPepGen/blob/e8a2cf5e55bd963bb1437d76100e45952d176cbf/moPepGen/aa/VariantPeptideIdentifier.py#L83-L88